protocol/nats_jetstream/v2/message.go

```diff
@@ -99,6 +99,17 @@ func (m *Message) ReadBinary(ctx context.Context, encoder binding.BinaryWriter)
 
 // Finish *must* be called when message from a Receiver can be forgotten by the receiver.
 func (m *Message) Finish(err error) error {
+	// Ignore error if message has already been Ack/Nak.
+	// Ack and Nak first checks to see if the message has been acknowleged
+	// and immediately returns an error without applying any logic to the message on the server.
+	// A message will auto acknowledge unless manual ack is set or policy is set to AckNonePolicy
+	// see: ManualAck() subscription option
+	// see: ConsumerConfig.AckPolicy
+	if err != nil {
+		_ = m.Msg.Nak()
```

Comment:
Does ignoring the return here potentially cause issues? Is the 'already ak'd' error the only kind of error that we can get from this operation?

Reply:
Added code to handle expected error cases.  In all other cases, will return error.